src/Illuminate/Bus/Dispatcher.php

```diff
@@ -44,16 +46,32 @@ class Dispatcher implements DispatcherContract, QueueingDispatcher, HandlerResol
 	protected $mapper;
 
 	/**
+	 * The pipeline
+	 *
+	 * @var Pipeline
+	 */
+	protected $pipeline;
+
+	/**
+	 * The pipes which the pipeline will send commands through
+	 *
+	 * @var array
+	 */
+	protected $pipes = [];
+
+	/**
 	 * Create a new command dispatcher instance.
 	 *
 	 * @param  \Illuminate\Contracts\Container\Container  $container
 	 * @param  \Closure|null $queueResolver
+	 * @param  \Illuminate\Contracts\Pipeline\Pipeline|null  $pipeline
 	 * @return void
 	 */
-	public function __construct(Container $container, Closure $queueResolver = null)
+	public function __construct(Container $container, Closure $queueResolver = null, PipelineContract $pipeline = null)
 	{
 		$this->container = $container;
 		$this->queueResolver = $queueResolver;
+		$this->pipeline = $pipeline ?: new Pipeline($container);
```

Comment:
thanks @GrahamCampbell -- I will make all the necessary changes.  I was just really putting this out there as a feeler to see if this would be something other users might be interested in.